winter/output_processor.py

```diff
@@ -20,30 +25,28 @@ class IOutputProcessorResolver(abc.ABC):
     """
 
     @abc.abstractmethod
-    def is_supported(self, body: typing.Any) -> bool:
+    def is_supported(self, body: typing.Any) -> bool:  # pragma: no cover
         return False
 
     @abc.abstractmethod
-    def get_processor(self, body: typing.Any) -> IOutputProcessor:
+    def get_processor(self, body: typing.Any) -> IOutputProcessor:  # pragma: no cover
         pass
 
 
 _registered_output_processors = {}
 _registered_resolvers: typing.List[IOutputProcessorResolver] = []
 
 
-def register_output_processor(func: typing.Callable, output_processor: IOutputProcessor):
-    if func in _registered_output_processors:
-        raise Exception(f'{func} already has registered output processor')
-    _registered_output_processors[func] = output_processor
+def register_output_processor(method: typing.Callable, output_processor: IOutputProcessor):
+    return annotate(output_processor, key=_key)(method)
```

Comment:
If you want to use annotations here, please create annotation for the purpose, for example:
```
@dataclass
class OutputProcessorAnnotation:
    output_processor: IOutputProcessor
```